workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/GameObjectWorldCommandSystem.cs

```diff
@@ -0,0 +1,155 @@
+                    foreach (var worldCommandHandler in worldCommandResponseHandlers)
+                    {
+                        worldCommandHandler.OnReserveEntityIdsResponseInternal(receivedResponse);
+                    }
+                }
+            }
+
+            for (var i = 0; i < createEntityResponseData.Length; ++i)
+            {
+                if (createEntityResponseData.Responses[i].Responses.Count == 0)
+                {
+                    continue;
+                }
+
+                var entity = createEntityResponseData.Entities[i];
+
+                var worldCommandResponseHandlers =
+                    GetWorldCommandResponseHandlersForEntity(entity);
+
+                if (worldCommandResponseHandlers == null)
+                {
+                    continue;
+                }
+
+                foreach (var receivedResponse in createEntityResponseData.Responses[i].Responses)
+                {
+                    foreach (var worldCommandHandler in worldCommandResponseHandlers)
+                    {
+                        worldCommandHandler.OnCreateEntityResponseInternal(receivedResponse);
+                    }
+                }
+            }
+
+            for (var i = 0; i < deleteEntityResponseData.Length; ++i)
+            {
+                if (deleteEntityResponseData.Responses[i].Responses.Count == 0)
+                {
+                    continue;
+                }
+
+                var entity = deleteEntityResponseData.Entities[i];
+
+                var worldCommandResponseHandlers =
+                    GetWorldCommandResponseHandlersForEntity(entity);
+
+                if (worldCommandResponseHandlers == null)
+                {
+                    continue;
+                }
+
+                foreach (var receivedResponse in deleteEntityResponseData.Responses[i].Responses)
+                {
+                    foreach (var worldCommandHandler in worldCommandResponseHandlers)
+                    {
+                        worldCommandHandler.OnDeleteEntityResponseInternal(receivedResponse);
+                    }
+                }
+            }
+        }
+
+        private WorldCommandsRequirables.WorldCommandResponseHandler[] GetWorldCommandResponseHandlersForEntity(
+            Entity entity)
+        {
+            var entityToReaderWriterStore = GameObjectDispatcherSystem.entityToReaderWriterStore;
+
+            if (!entityToReaderWriterStore.TryGetValue(entity, out var injectableStore))
+            {
+                return null;
+            }
+
+            if (!injectableStore.TryGetInjectablesForComponent(WorldCommandResponseHandlerInjectableId,
+                out var injectables))
+            {
+                return null;
+            }
+
+            if (injectables.Count == 0)
+            {
+                return null;
+            }
+
+            return injectables.Cast<WorldCommandsRequirables.WorldCommandResponseHandler>().ToArray();
```

Comment:
??

Reply:
It's an expensive way to do that. Anything that uses IEnumerable shouldn't be used in a hot path.

Reply:
Will go for
```cs
Array.ConvertAll(injectables.ToArray(),
                injectable => (WorldCommandRequirables.WorldCommandResponseHandler) injectable)
```

Reply:
So that updated version will turn a HashSet into an array and then map it to another? Still could be lighter by just iterating over the HashSet and casting each element I think, I don't have a strong opinion about it though

Reply:
Really the hashset probably shouldn't be a hashset. The majority of operations on it are iterating over it.

Reply:
The other option (which I think I prefer longer term) is to do this without the type erasure. However that is probably too large a change for now.

Reply:
Creating a ticket for this, will link

Reply:
UTY-1146

Reply:
It's a hashset at the moment because we remove arbitrary elements from it as monobehaviours are disabled. But thinking about it more, that's not a good reason if we iterate much more often than we insert and don't have loads of readers/etc of the same type for a single entity, ie it should just be a list.